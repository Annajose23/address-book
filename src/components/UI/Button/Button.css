.button {
    border: 1px solid #eee;
    text-align: center;
    background-color: #3A3B3C;
    color:white;
    padding: 16px;
    border-radius: 5px;
}