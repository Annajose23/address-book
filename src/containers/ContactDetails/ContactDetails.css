.contact-details {
    width: 100%;
    margin: 30px auto;
    border: 1px solid #eee;
    text-align: center;
    border-radius: 10px;
    background-color: #3A3B3C;
}

.contact-details + div{
    display: flex;
    flex-flow: row wrap;
    justify-content: center;
}

@media(min-width:40rem){
    .contact-details{
        width: 70%;
    }
}