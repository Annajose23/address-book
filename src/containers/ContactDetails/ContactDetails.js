import React from "react";
import { connect } from "react-redux";

import Aux from "../../hoc/Auxilary/Auxilary";
import Button from '../../components/UI/Button/Button';
import "./ContactDetails.css";

export const ContactDetails = (props) => {
   
  const clickedHandler = () => {
    props.history.replace('/');
  };

  return (
    props.selectedContact && (
      <Aux>
        <div className="contact-details">
          <h3>Contact Details</h3>
          <h2>
            {props.selectedContact.name.title}.{" "}
            {props.selectedContact.name.first} {props.selectedContact.name.last}
          </h2>
          <p>Phone number: {props.selectedContact.cell}</p>
        </div>
        <div >
          <Button clicked={clickedHandler}>Home</Button>
        </div>
      </Aux>
    )
  );
};

const mapStateToProps = (state) => {
  return {
    selectedContact: state.selectedContact,
  };
};

export default connect(mapStateToProps)(ContactDetails);
