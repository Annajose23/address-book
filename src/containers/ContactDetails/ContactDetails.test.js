import React from "react";
import renderer from "react-test-renderer";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { ContactDetails } from "./ContactDetails";
import Button from "../../components/UI/Button/Button";
import Aux from "../../hoc/Auxilary/Auxilary";

configure({ adapter: new Adapter() });

describe("<ContactDetails />", () => {
  let wrapper;

  let props = {
    selectedContact: {
      name: {
        title: "Ms",
        first: "Anna",
        last: "Jose",
      },
      cell: "0123456",
    },
  };
  
  beforeEach(() => {
    wrapper = shallow(<ContactDetails {...props} />);
  });

  it("should render <Button /> when receiving selectedContact", () => {
    expect(wrapper.find(Button)).toHaveLength(1);
  });

  it("should contain div tag", () => {
    expect(wrapper.find("div")).toHaveLength(2);
  });

  it("should contain h3 and h2 tag", () => {
    expect(wrapper.find("h3")).toHaveLength(1);
    expect(wrapper.find("h2")).toHaveLength(1);
  });

  it("should contain Aux", () => {
    expect(wrapper.find(Aux)).toHaveLength(1);
  });

  it("render correctly Contact Details component", () => {
    const ContactDetailsComponent = renderer
      .create(<ContactDetails {...props}/>)
      .toJSON();
    expect(ContactDetailsComponent).toMatchSnapshot();
  });
});
