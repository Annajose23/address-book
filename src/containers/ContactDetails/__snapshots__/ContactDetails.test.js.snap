// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactDetails /> render correctly Contact Details component 1`] = `
Array [
  <div
    className="contact-details"
  >
    <h3>
      Contact Details
    </h3>
    <h2>
      Ms
      .
       
      Anna
       
      Jose
    </h2>
    <p>
      Phone number: 
      0123456
    </p>
  </div>,
  <div>
    <button
      className="button"
      onClick={[Function]}
    >
      Home
    </button>
  </div>,
]
`;
